pkg/reconciler/common/workload_override.go

```diff
@@ -71,6 +75,13 @@ func OverridesTransform(overrides []base.WorkloadOverride, log *zap.SugaredLogge
 				ps = &job.Spec.Template
 			}
 
+			if u.GetKind() == "HorizontalPodAutoscaler" && u.GetName() == override.Name && override.Replicas != nil {
```

Comment:
Current `[]base.WorkloadOverride` (`spec.workloads`) assume that it contains only `Deployment`, `StatefulSet`, and `Job`, so this condition does not happen. 

We **may** able to support PDB for `spec.workloads` but it is weird that users need to set `override.Replicas` for PDB.

Reply:
Ah, I guess the code assume `Deployment`'s name is same with `PDB`'s name? If so, it is different like https://github.com/knative/serving/blob/ba5a9298901945907498c7e6b9c28a09bc346f59/config/core/deployments/webhook-hpa.yaml#L44

Reply:
Ah, the code assumes that `Deployment` name is same with `HorizontalPodAutoscaler`'s name?

Reply:
Yes, AFAIK this is the current assumption of the operator. This PR just introduces the same same behaviour as for the `high-availability` flag. So the example for `mt-channel-broker` would also currently not work with `high-availability` set.

We probably should introduce a new concept to configure the HPAs with min+max fields anyways, but that is another story (mid to long term). So for the short term, can eventing change the name of the HPA? @matzew , @pierDipi

Reply:
Changing the name requires some form of migration, can't we have a mapping `deployment` -> `hpa name` and use that?

Reply:
Let's do a follow up PR to fix the inconsistencies with the names.

Reply:
> Yes, AFAIK this is the current assumption of the operator. This PR just introduces the same same behaviour as for the high-availability flag. 

I think `high-availability` (=`HighAvailabilityTransform()`) does not have the assumption. It applies min/max replica settings to all HPA regardless of the name.
(Note `hasHorizontalPodAutoscaler()` is to avoid another issue, not for the inconsistent name.)

Reply:
I agree with you from the code, but it still has a bit the assumption, as `hasHorizontalPodAutoscaler` would currently interfere with the eventing HPAs, so I added the mapping @pierDipi proposed. Now all three cases should be fine:

**Given** 
- mt-broker-ingress: replicas are not set in yaml
- broker-ingress-hpa: 1-10

**Case 1** 
- high-available=1
- no workload overrides

HighAvailabilityTransform
- mt-broker-ingress —> skipped
- broker-ingress-hpa —> not skipped, _set to 1_

OverridesTransform
- mt-broker-ingress —> skipped
- broker-ingress-hpa —> skipped

**Case 2**
- high-available not set
- workload overrides: mt-broker-ingress=2

HighAvailabilityTransform
- mt-broker-ingress —> skipped
- broker-ingress-hpa —> skipped

OverridesTransform
- mt-broker-ingress —> _set to 2_
- broker-ingress-hpa —> _set to 2_


**Case 3**
- high-available=3
- workload overrides: mt-broker-ingress=2

HighAvailabilityTransform
- mt-broker-ingress —> skipped
- broker-ingress-hpa —> not skipped, _set to 3_

OverridesTransform (this is executed later, as it is more specific)
- mt-broker-ingress —> _set to 2_
- broker-ingress-hpa —> _set to 2_

Reply:
Are these all covered in tests?

Reply:
I think so, but I'll double-check.

Reply:
Thanks, followup is fine as well

Reply:
> Are these all covered in tests?

Now all those cases are covered in tests.